High Throughput Sequencing technologies and bioinformatics analysis fall 2016
==========================================

Note: for previous course editions, check out [this website](http://inf-biox121.readthedocs.io).

This is the webpage for the fall 2016 edition of the credited courses [**INF-BIO5121**](http://www.uio.no/studier/emner/matnat/ifi/INF-BIO5121) (master level) and [**INF-BIO9121**](http://www.uio.no/studier/emner/matnat/ifi/INF-BIO9121/) (Ph.D. level) offered by the Department of Informatics and the Department of Biosciences at the University of Oslo (UiO). Students taking the course for credit should register through UiO StudentWeb. Non UiO students are welcome and should check [this website](http://www.uio.no/english/studies/admission/).

The High Throughput Sequencing technologies and bioinformatics
analysis course consists of five weeks, three days each week, of
lectures and practicals, and a final take-home exam, plus a written exam on the course material and the reading material.

## Schedule
Course days are from 9:00 to 16:00 (teachers may stay longer if
requested), some lectures, mostly hands-on exercises. All materials for lectures and practicals will be linked from the schedule below. 

The schedule links to the webpages with the material taught (these links are added as the course progresses).

<table border="1">
<thead>
<tr class="header">
<th align="left">Day</th>
<th align="left">Room</th>
<th align="left">Morning (09:00-12:00)</th>
<th align="left">Afternoon (13:00-16:00)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left"><strong>Week 34</strong></td>
</tr>
<tr class="even">
<td align="left">Tuesday August 23rd</td>
<td align="left"><strong>Smalltalk</strong></td>
<td align="left">Unix pre course</td>
<td align="left">Unix pre course</td>
</tr>
<tr class="odd">
<td align="left"><strong>Week 36</strong></td>
</tr>
<tr class="even">
<td align="left">Wednesday Sept. 7</td>
<td align="left"><strong>C</strong></td>
<td align="left"><a href="Intro_HTS/index.html">High-throughput sequencing</a></td>
<td align="left"><a href="Intro_HTS/index.html">High-throughput sequencing</a></td>
</tr>
<tr class="odd">
<td align="left">Thursday Sept. 8</td>
<td align="left">Python</td>
<td align="left"><a href="Intro_HTS/index.html">Principles and problems of HTS data analysis</a></td>
<td align="left"><a href="Intro_HTS/index.html">Experimental design</a></td>
</tr>
<tr class="even">
<td align="left">Friday Sept. 9</td>
<td align="left">Python</td>
<td align="left"><a href="QC/Read_QC.html">Read QC + trimming</a></td>
<td align="left"><a href="Intro_HTS/index.html">Project and data organisation, reporting</a></td>
</tr>
<tr class="odd">
<td align="left"><strong>Week 37</strong></td>
</tr>
<tr class="even">
<td align="left">Monday Sept. 12</td>
<td align="left">Python</td>
<td align="left"><a href="Assembly/index.html"><em>De novo</em> assembly</a></td>
<td align="left"><a href="Assembly/index.html"><em>De novo</em> assembly</a></td>
</tr>
<tr class="odd">
<td align="left">Tuesday Sept. 13</td>
<td align="left">Python</td>
<td align="left"><a href="Assembly/index.html"><em>De novo</em> assembly</a></td>
<td align="left"><a href="Assembly/index.html"><em>De novo</em> assembly</a></td>
</tr>
<tr class="even">
<td align="left">Wednesday Sept. 14</td>
<td align="left">Python</td>
<td align="left"><a href="Assembly/index.html"><em>De novo</em> assembly</a></td>
<td align="left"><a href="Assembly/index.html"><em>De novo</em> assembly</a></td>
</tr>
<tr class="odd">
<td align="left"><strong>Week 38</strong></td>
</tr>
<tr class="even">
<td align="left">Monday Sept. 19</td>
<td align="left">Python</td>
<td align="left"><a href="RNAseq/index.html">RNA seq: differential expression analysis</a></td>
<td align="left"><a href="RNAseq/index.html">RNA seq: differential expression analysis</a></td>
</tr>
<tr class="odd">
<td align="left">Tuesday Sept. 20</td>
<td align="left"><strong>Java</strong></td>
<td align="left"><a href="RNAseq/index.html">RNA seq: differential expression analysis</a></td>
<td align="left"><a href="RNAseq/index.html">RNA seq: differential expression analysis</a></td>
</tr>
<tr class="even">
<td align="left">Wednesday Sept. 21</td>
<td align="left"><strong>Smalltalk</strong></td>
<td align="left"><a href="RNAseq/index.html">RNA seq: differential expression analysis</a></td>
<td align="left"><a href="RNAseq/index.html">RNA seq: differential expression analysis</a></td>
</tr>
<tr class="odd">
<td align="left"><strong>Week 39</strong></td>
</tr>
<tr class="even">
<td align="left">Wednesday Sept. 28</td>
<td align="left">Python</td>
<td align="left">Mapping and variant calling</td>
<td align="left">Mapping and variant calling</td>
</tr>
<tr class="odd">
<td align="left">Thursday Sept. 29</td>
<td align="left">Python</td>
<td align="left">Mapping and variant calling</td>
<td align="left">Mapping and variant calling</td>
</tr>
<tr class="even">
<td align="left">Friday Sept. 30</td>
<td align="left">Python</td>
<td align="left">Mapping and variant calling</td>
<td align="left">Mapping and variant calling</td>
</tr>
<tr class="odd">
<td align="left"><strong>Week 40</strong></td>
</tr>
<tr class="even">
<td align="left">No teaching, 'høstferie'</td>
</tr>
<tr class="odd">
<td align="left"><strong>Week 41</strong></td>
</tr>
<tr class="even">
<td align="left">Monday Oct. 10</td>
<td align="left">Python</td>
<td align="left">Reproducible bioinformatics using Galaxy</td>
<td align="left">Reproducible bioinformatics using Galaxy</td>
</tr>
<tr class="odd">
<td align="left">Tuesday Oct. 11</td>
<td align="left">Python</td>
<td align="left">Statistical genomics (Hyperbrowser)</td>
<td align="left">Statistical genomics (Hyperbrowser)</td>
</tr>
<tr class="even">
<td align="left">Wednesday Oct. 12</td>
<td align="left">Python</td>
<td align="left">Statistical genomics (Hyperbrowser)</td>
<td align="left">Statistical genomics (Hyperbrowser)</td>
</tr>
<tr class="odd">
<td align="left"><strong>Week 42</strong></td>
</tr>
<tr class="even">
<td align="left">home exam</td>
</tr>
<tr class="odd">
<td align="left"><strong>Week 43</strong></td>
</tr>
<tr class="even">
<td align="left">home exam</td>
</tr>
<tr class="odd">
<td align="left"><strong>Week 44 </strong></td>
</tr>
<tr class="even">
<td align="left">Thursday Nov. 3</td>
<td align="left">TBD</td>
<td align="left">Oral examination of the home exam, 30 minutes</td>
<td align="left">Oral examination of the home exam, 30 minutes</td>
</tr>
<tr class="odd">
<td align="left">Friday Nov. 4</td>
<td align="left">TBD</td>
<td align="left">Written exam 9:00-11:00</td>
</tr>
</tbody>
</table>
<p><p>

**Room locations:**

* Smalltalk: room 1416, first floor auditorium near the man entrance  (Ole-Johan Dahls hus)
* Python: room 2269 Informatics building (Ole-Johan Dahls hus)
* Java: room 2423 Informatics building (Ole-Johan Dahls hus)
* KN: 'Lille auditorium' in Kristin Nygaards hus (next to the Informatics building)
* C: room 3437 Informatics building (Ole-Johan Dahls hus)

## Requirements

* No prior background in bioinformatics or computer science is required, however, we expect students to have a basic understanding of the unix shell. For this, we organise an **introductory unix course** about a
week before the course starts - details will be provided upon acceptance. Participants who are familiar with the shell
can ask to be exempt for this course day.
* All participants should have a basic understanding of molecular
biology, at least roughly corresponding to 5-10 university study points
in molecular biology, biochemistry, or similar. *If you are uncertain if
your biology background is strong enough, please contact the course
coordinators (see contact details below) at least three weeks before the
start of the course.*


## Computers/laptops, internet access, and UiO user account
All students must bring a laptop with either a Windows (Windows 7 or more recent), Unix/Linux, or OS X (i.e. an Apple computer) operating system.

* The computer should not be more than 2-3 years old
* It should be possible to connect the computer to the UiO wireless network
* You must have modern internet browser installed (Chrome, Firefox, Safari, ...)
* You must have a valid UiO user account and must be able to log onto a computer on the UiO network
  * If you are unsure if you have a UiO user account and a valid password, you should try to log in using kiosk.uio.no as described [here](http://www.uio.no/english/services/it/network/home-and-away/kiosk/index.html). If you are unable to log in, try the hints you find [here](http://www.uio.no/english/services/it/network/home-and-away/kiosk/programkiosk/index.html).
  * Instructions (in Norwegian) about how to find your user name and get a new password can be found [here](http://www.uio.no/tjenester/it/brukernavn-passord/ikke-passord.html).
* We advise to bring an external mouse, and do not rely on touchpad/trackpad only

If you are struggling with anything of the above, in particular if you have forgotten your UiO user name/password or you do not have one, you must contact the course coordinator (see contact details on this page) as soon as possible, and at least one week before the start of the course.

## Curriculum and suggested reading
The curriculum consists of a set of scientific articles. The exam will
contain questions on the material presented in these papers on the material covered during the course. Specific details of the curriculum, and a list of further suggested papers, can be found [here](misc/curriculum.html).

## Exam

The exam for this course will be:

-   a written exam on the course and curriculum, time and place
    to be announced
-   an *individual*, home exam
    combined with an oral examination of the home exam time, and place
    to be announced. Students will present their work to two-three teachers,
    followed by some questioning (20-30 minutes in total). This home exam will
    be handed out to all participants during the last course day.
    

**NOTE** Both exams have to be passed to earn 10 study credits.
  
**NOTE** 80% attendance is required to take the final exam.

### Exams from previous years
* [2015](https://github.com/lexnederbragt/INF-BIOx121/tree/2015/exam) (with information on how the grading was done)
* [2014](https://wiki.uio.no/projects/clsi/images/f/f2/Examinfbio2014.pdf) (exam only)

## Contact information

* Lex Nederbragt (Course coordinator) - e-mail: lex.nederbragt - at -
ibv.uio.no
* Karin Lagesen (Course co-coordinator) - e-mail: karin.lagesen - at -
medisin.uio.no
* Course administration/registration - e-mail: studieinfo@ifi.uio.no

## Teachers

* Lex Nederbragt (LN, Centre for Ecological and Evolutionary Synthesis,
Dept. of Biosciences & Research Group for Biomedical Informatics Dept. of Informatics, University of Oslo)
* Timothy Hughes (TH, Researcher, NORMENT , Oslo Univ. Hospital)
* Monica Solbakken (MS, Centre for Ecological and Evolutionary Synthesis,
Dept. of Biosciences, University of Oslo)
* Rebekah Oomen (RO, Centre for Ecological and Evolutionary Synthesis,
Dept. of Biosciences, University of Oslo)
* Boris Simovski (BS, Research Group for Biomedical Informatics, Dept. of
Informatics, University of Oslo)
* Karin Lagesen (KL, Norwegian Sequencing Centre, Oslo Univ.
Hospital/Dept. of Informatics)
