Genome Sequencing Technologies, Assembly, Variant Calling and Statistical Genomics fall 2020  
==========================================
IN-BIOS5000 and IN-BIOS9000
==========================================

Note: for previous course editions (INF-BIO9121 and INF-BIO5121), check out [this website](http://inf-biox121.readthedocs.io).

## Lecture slides

[Lecture slides](Lectures)<br/>
This page is updated after each lecture/Module.

## Teaching materials

Link to [Etherpad](https://etherpad.wikimedia.org/p/INF_BIOSx000_2020)<br/>
List of further suggested papers, can be found [here](Curriculum).  

## Information on compute
Link to [Procedure](Information)<br/>

## Home exam

Link to [home exam](misc)

## Introduction

This is the webpage for the fall 2020 edition of the credited courses [**IN-BIOS5000**](http://www.uio.no/studier/emner/matnat/ifi/IN-BIOS5000) (master level) and [**IN-BIOS9000**](http://www.uio.no/studier/emner/matnat/ifi/IN-BIOS9000) (Ph.D. level) offered by the Department of Informatics and the Department of Biosciences at the University of Oslo (UiO) and funded by [**NORBIS**](https://norbis.w.uib.no/activities/courses/genome-sequencing-technologies-assembly-variant-calling-and-statistical-genomics).

ThisÂ course consists of two weeks, of
lectures and practicals, and a final take-home exam, plus a written exam on the course material and the reading material.

## Schedule
Course days are usually from 9:15 to 17:00 (teachers may stay longer if
requested), some lectures, mostly hands-on exercises. All materials for lectures and practicals will be linked from the schedule below.

The schedule will appear here early in August and will link to the webpages with the material taught (these links are added as the course progresses).


**Room location:**

<table border="1">
<thead>
<tr class="header">
<th align="left">Day</th>
<th align="left">Time</th>
<th align="left">Session</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left"><strong>Week 43</strong></td>
</tr>
<tr class="even">
<td align="left">Mon. Oct. 19</td>
<td align="left">09:15-09:30<br/>09:30-12:00<br/>13:00-14:30<br/>14:45-15:30<br/>15:45-16:30</td>
<td align="left">Update on NORBIS (RB)<br/>Fundamentals of Molecular Biology (GG)<br/>HTS & applications (TR) (pre-recorded)<br/>Illumina sequencing (AS)<br/>PacBio technology (AT)</td>
</tr>
<tr class="odd">
<td align="left">Tue. Oct. 20</td>
<td align="left">09:15-10:00<br/>10:15-11:00<br/>11:15-12:00<br/>13:00-13:45<br/>14:00-16:00</td>
<td align="left">Oxford Nanopore technology (THa) (pre-recorded)<br/>Library preparation (AS)<br/>Experimental design (AS)<br/>Access to UiO and compute<br/>Basic file formats (AS)</td>
</tr>
<tr class="even">
<td align="left">Wed. Oct. 21</td>
<td align="left">09:15-12:00<br/>13:00-17:00</td>
<td align="left">Pre-processing of HTS data (AS)<br/>Algorithms (THu)</td>
</tr>
<tr class="odd">
<td align="left">Thu. Oct. 22</td>
<td align="left">09:15-12:00<br/>13:00-16:00</td>
<td align="left">Algorithms (THu)<br/>De novo assembly (KL)</td>
</tr>
<tr class="even">
<td align="left">Fri. Oct. 23</td>
<td align="left">09:15-16:00</td>
<td align="left">De novo assembly (KL)</td>
</tr>
<tr class="odd">
<td align="left"><strong>Week 44</strong></td>
</tr>
<tr class="even">
<td align="left">Mon. Oct. 26</td>
<td align="left">09:15-16:00</td>
<td align="left">De novo assembly (KL)</td>
</tr>
<tr class="odd">
<td align="left">Tue. Oct. 27</td>
<td align="left">09:15-16:00</td>
<td align="left">Statistical genomics (IG)</td>
</tr>
<tr class="even">
<td align="left">Wed. Oct. 28</td>
<td align="left">09:15-16:00</td>
<td align="left">Statistical genomics (IG)</td>
</tr>
<tr class="odd">
<td align="left">Thu. Oct. 29</td>
<td align="left">09:15-16:00</td>
<td align="left">Introduction to machine learning in computational biology (MP)</td>
</tr>
<tr class="even">
<td align="left">Fri. Oct. 30</td>
<td align="left">09:15-12:00<br/>13:00-14:00<br/>14:15-15:15<br/>15:30-1600</td>
<td align="left">Small RNA transcriptomics (TBR)<br/>Introduction to somatic variant calling (DV)<br/>Interpretation of DNA variation in the context of cancer (SN)<br/>Course evaluation</td>
</tr>
<tr class="odd">
<td align="left"><strong>Week 48</strong></td>
</tr>
<tr class="even">
<td align="left">Thu. Nov. 26</td>
<td align="left">TBA</td>
<td align="left">Oral exam</td>
</tr>
<tr class="odd">
<td align="left">Fri. Nov. 27</td>
<td align="left">TBA</td>
<td align="left">Oral exam</td>
</tr>
 </tbody>
</table>
<p><p>
 
<!---
<table border="1">
<thead>
<tr class="header">
<th align="left">Day</th>
<th align="left">Room</th>
<th align="left">Morning (09:15-12:00)</th>
<th align="left">Afternoon (13:00-17:00)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left"><strong>Week 43</strong></td>
</tr>
<tr class="even">
<td align="left">Mon. Oct. 19</td>
<td align="left">XXX</td>
<td align="left">Update on NORBIS (RB)<br/>Fundamentals of Molecular Biology (GG)</td>
<td align="left">HTS & applications (TR)<br/>Illumina sequencing (AS)<br/>PacBio technology (AT)</td>
</tr>
<tr class="odd">
<td align="left">Tue. Oct. 20</td>
<td align="left">XXX</td>
<td align="left">Oxford Nanopore technology (THa)<br/>Library preparation (AS)<br/>Experimental design (AS)</td>
<td align="left">Basic file formats (AS)</td>
</tr>
<tr class="even">
<td align="left">Wed. Oct. 21</td>
<td align="left">XXX</td>
<td align="left">Pre-processing of HTS data (AS)</td>
<td align="left">Algorithms (THu)</td>
</tr>
<tr class="odd">
<td align="left">Thu. Oct. 22</td>
<td align="left">XXX</td>
<td align="left">Algorithms (THu)</td>
<td align="left">De novo assembly (KL)</td>
</tr>
<tr class="even">
<td align="left">Fri. Oct. 23</td>
<td align="left">XXX</td>
<td align="left">De novo assembly (KL)</td>
<td align="left">De novo assembly (KL)</td>
</tr>
<tr class="odd">
<td align="left"><strong>Week 44</strong></td>
</tr>
<tr class="even">
<td align="left">Mon. Oct. 26</td>
<td align="left">XXX</td>
<td align="left">De novo assembly (KL)</td>
<td align="left">De novo assembly (KL)</td>
</tr>
<tr class="odd">
<td align="left">Tue. Oct. 27</td>
<td align="left">XXX</td>
<td align="left">Statistical genomics (IG)</td>
<td align="left">Statistical genomics (IG)</td>
</tr>
<tr class="even">
<td align="left">Wed. Oct. 28</td>
<td align="left">XXX</td>
<td align="left">Statistical genomics (IG)</td>
<td align="left">Statistical genomics (IG)</td>
</tr>
<tr class="odd">
<td align="left">Thu. Oct. 29</td>
<td align="left">XXX</td>
<td align="left">Introduction to machine learning in computational biology (MP)</td>
<td align="left">Introduction to machine learning in computational biology (MP)</td>
</tr>
<tr class="even">
<td align="left">Fri. Oct. 30</td>
<td align="left">XXX</td>
<td align="left">Small RNA transcriptomics (TBR)</td>
<td align="left">Somatic variant calling (DV)</td>
</tr>
<tr class="odd">
<td align="left"><strong>Week 48</strong></td>
</tr>
<tr class="even">
<td align="left">Thu. Nov. 26</td>
<td align="left">TBA</td>
<td align="left">Oral exam</td>
<td align="left">Oral exam</td>
</tr>
<tr class="odd">
<td align="left">Fri. Nov. 27</td>
<td align="left">TBA</td>
<td align="left">Oral exam</td>
<td align="left">Oral exam</td>
</tr>
 </tbody>
</table>
<p><p>
--->

**Room location:**

All activities will be carried out on Zoom. Zoom info will be distributed by email.
<!---
Ole-Johan Dahls hus location: <a href='https://goo.gl/maps/b1qoR'>Google maps</a>
--->
<!---
**Dinner on Oct XX**
--->

**Exam location:**
<!---
Oral exam on week 48; Nov 26 and 27. Home exam.
--->
Exam: Oral presentation of home exam and questioning on Nov 26 and 27 by Zoom. Zoom info will be distributed by email.

## Requirements

* We recommend that participants should have a basic understanding of molecular biology as provided by an introductory course in bioinformatics, molecular biology, or genetics.

* No formal background in computer science is required, however, students must have a basic understanding of the unix shell. Students should take an introductory unix course beforehand if they do not have these skills.


## Computers/laptops, internet access, and UiO user account
All students must have a laptop with either a Windows (Windows 7 or more recent), Unix/Linux, or OS X (i.e. an Apple computer) operating system.

* The computer should not be more than 2-3 years old
<!---* It should be possible to connect the computer to the UiO wireless network--->
* You must have modern internet browser installed (Chrome, Firefox, Safari, ...)
* You must have a valid UiO user account and must be able to log onto a computer on the UiO network
  * If you are unsure if you have a UiO user account and a valid password, you should try to log in using kiosk.uio.no as described [here](http://www.uio.no/english/services/it/network/home-and-away/kiosk/index.html). If you are unable to log in, try the hints you find [here](http://www.uio.no/english/services/it/network/home-and-away/kiosk/programkiosk/index.html).
  * Instructions (in Norwegian) about how to find your user name and get a new password can be found [here](http://www.uio.no/tjenester/it/brukernavn-passord/ikke-passord.html).
* We advise to bring an external mouse, and do not rely on touchpad/trackpad only

If you are struggling with anything of the above, in particular if you have forgotten your UiO user name/password or you do not have one, you must contact the course coordinator (see contact details on this page) as soon as possible, and at least one week before the start of the course.

## Curriculum and suggested reading
The curriculum consists of a set of scientific articles. The exam will
contain questions on the material presented in these papers on the material covered during the course. Specific details of the curriculum, and a list of further suggested papers, can be found [here](Curriculum).

## Exam

The exam for this course will be an *individual* home exam combined with an oral presentation of the home exam and questioning on Zoom on 26/27 November. Exact time to be announced. Students will present their work to two-three teachers, followed by some questioning (20-30 minutes in total). This home exam will be handed out to all participants during the last course day.

**NOTE** Both exams have to be passed to earn study credits.

**NOTE**Â 80% attendance is required to take the final exam.

### Written exams from previous years
* [2018](https://github.com/arvindsundaram/IN-BIOSx000/blob/2018/misc/written_exam.pdf)
* [2016](https://github.com/lexnederbragt/INF-BIOx121/blob/2017/misc/2016.pdf) (exam only)
* [2015](https://github.com/lexnederbragt/INF-BIOx121/tree/2015/exam) (with information on how the grading was done)
* [2014](https://wiki.uio.no/projects/clsi/images/f/f2/Examinfbio2014.pdf) (exam only)

## Contact information

* Arvind Sundaram (Course coordinator) - e-mail: arvind.sundaram - at -medisin.uio.no
* Course administration/registration - e-mail: studieinfo@ifi.uio.no

## Teachers

* Arvind Sundaram (AS, Bioinformatician, Norwegian Sequencing Centre, Oslo University Hospital)
* Ave Tooming-Klunderud (AT, CEES, Faculty of Mathematics and Natural Sciences, University of Oslo)
* Daniel Vodak (DV, Oslo University Hospital)
* Gregor Gilfillan (GG, Researcher, Norwegian Sequencing Centre, Oslo University Hospital)
* Ivar Grytten (IG, Research Group for Biomedical Informatics, Dept. of
Informatics, University of Oslo)
* Karin Lagesen (KL, Section for Epidemiology, Norwegian Veterinary Institute)
* Milena Pavlovic (MP, Dept. of Informatics, University of Oslo)
* Ragna Breines (RB, Coordinator, NORBIS)
* Sigve Nakken (SN, Post.Doc., Oslo University Hospital)
* Thomas Haverkamp (THa, Section for Epidemiology, Norwegian Veterinary Institute)
* Timothy Hughes (THu, Researcher, NORMENT , Oslo University Hospital)
* TorbjÃ¸rn Rognes (TR, Research Group for Biomedical Informatics, Dept. of
Informatics, University of Oslo)
* Trine B. Rounge (TBR, Researcher, Kreftregisteret)
