
# Curriculum 2020

##### Molecular Biology
Refer to Wikipedia for basic information

##### High throughput sequencing (HTS)
[The sequence of sequencers: The history of sequencing DNA](https://github.com/arvindsundaram/IN-BIOSx000/raw/2020/Curriculum/HTS_history.pdf), Heather JM _et. al._, Genomics 2016

[High-Throughput Sequencing Technologies](https://github.com/arvindsundaram/IN-BIOSx000/raw/2020/Curriculum/HTS_technology.pdf), Reuter JA _et. al._, Molecular Cell 2015

[Sequencing depth and coverage: key considerations in genomic analyses](https://github.com/arvindsundaram/IN-BIOSx000/raw/2020/Curriculum/HTS_Coverage_Depth.pdf), Sims D _et. al._, Nature Reviews 2014

[RNA-seq differential expression studies: more sequence or more replication?](https://github.com/arvindsundaram/IN-BIOSx000/raw/2020/Curriculum/HTS_DE.pdf), Liu Y _et. al._, Bioinformatics 2014

##### _De novo_ genome assembly
[A field guide to whole‐genome sequencing, assembly and annotation](https://wiki.uio.no/projects/clsi/images/9/99/Ekblom2014.pdf), Ekblom and Wolf, Evolutionary Applications 2014

[The Theory and Practice of Genome Sequence Assembly](https://wiki.uio.no/projects/clsi/images/8/8f/Simpson2015.pdf), Simpson and Pop, Annual Review of Genomics and Human Genetics 2015

##### Variant Calling

[Genotype and SNP calling from next-generation sequencing data](https://github.com/arvindsundaram/IN-BIOSx000/raw/2018/Curriculum/nrg2986.pdf) Nielsen et al., Nature Reviews Genetics 2011. This is a review paper that should be easily understandable once the student has taken the variant calling module. 

[Exome sequencing identifies the cause of a mendelian disorder](https://github.com/arvindsundaram/IN-BIOSx000/raw/2018/Curriculum/ng.499.pdf) Ng et al., Nature Genetics 2010. A paper reporting the first resolution of a mendelian disorder using exome capture.

##### Statistical genomics
[Colocalization analyses of genomic elements: approaches, recommendations and challenges, Kanduri et. al.]((https://github.com/arvindsundaram/IN-BIOSx000/raw/2020/Curriculum/ST_1.pdf), Bioinformatics 2018.

[Simovski, Boris, et al. "GSuite HyperBrowser: integrative analysis of dataset collections across the genome and epigenome."](https://github.com/arvindsundaram/IN-BIOSx000/raw/2020/Curriculum/ST_2.pdf)

[Quinlan, Aaron R., and Ira M. Hall. "BEDTools: a flexible suite of utilities for comparing genomic features."](https://github.com/arvindsundaram/IN-BIOSx000/raw/2020/Curriculum/ST_3.pdf) Bioinformatics 26.6 (2010): 841-842.

[Optional reading for those interested: Salvatore, Stefania, et al.](https://github.com/arvindsundaram/IN-BIOSx000/raw/2020/Curriculum/ST_4.pdf) "Beware the Jaccard: the choice of similarity measure is important and non-trivial in genomic colocalisation analysis." Briefings in bioinformatics 21.5 (2020): 1523-1530
