
# Curriculum

##### Molecular Biology
Refer to Wikipedia for basic information

##### High throughput sequencing (HTS)
[The sequence of sequencers: The history of sequencing DNA](https://github.com/lexnederbragt/INF-BIOx121/raw/2017/Curriculum/HTS_history.pdf), Heather JM _et. al._, Genomics 2016

[High-Throughput Sequencing Technologies](https://github.com/lexnederbragt/INF-BIOx121/raw/2017/Curriculum/HTS_technology.pdf), Reuter JA _et. al._, Molecular Cell 2015

[Sequencing depth and coverage: key considerations in genomic analyses](https://github.com/lexnederbragt/INF-BIOx121/raw/2017/Curriculum/HTS_Coverage_Depth.pdf), Sims D _et. al._, Nature Reviews 2014

[RNA-seq differential expression studies: more sequence or more replication?](https://github.com/lexnederbragt/INF-BIOx121/raw/2017/Curriculum/HTS_DE.pdf), Liu Y _et. al._, Bioinformatics 2014

##### _De novo_ genome assembly

##### RNA-seq analysis
[A survey of best practices for RNA-seq data analysis](https://github.com/lexnederbragt/INF-BIOx121/raw/2017/Curriculum/RNAseq_BestPractice.pdf), Conesa A _et. al._, Genome Biology 2016 

[Differential gene and transcript expression analysis of RNA-seq experiments with TopHat and Cufflinks](https://github.com/lexnederbragt/INF-BIOx121/raw/2017/Curriculum/RNAseq_Tophat.pdf), Trapnell C _et. al._, Nature Protocol 2014 

[De novo transcript sequence reconstruction from RNA-seq using the Trinity platform for reference generation and analysis](https://github.com/lexnederbragt/INF-BIOx121/raw/2017/Curriculum/RNAseq_Trinity.pdf), Hass BJ _et. al._, Nature Protocol 2013

[Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2](https://github.com/lexnederbragt/INF-BIOx121/raw/2017/Curriculum/RNAseq_DESeq2.pdf), Love MI _et. al._, Genome Biology 2014 

[edgeR: a Bioconductor package for differential expression analysis of digital gene expression data](https://github.com/lexnederbragt/INF-BIOx121/raw/2017/Curriculum/RNAseq_edgeR.pdf), Robinson MD _et. al._, Bioinformatics 2010 

##### Variant Calling

##### Statistical genomics

[Ten Simple Rules for Reproducible Computational Research](https://wiki.uio.no/projects/clsi/images/e/ed/Sandve2013-2.pdf), Sandve _et. al._, PLoS Computational Biology 2013

[The dilemma of choosing the ideal permutation strategy while estimating statistical significance of genome-wide enrichment](https://wiki.uio.no/projects/clsi/images/9/92/The_dilemma_of_choosing_the_ideal_permutation_strategy_while_estimating_statistical_significance_of_genome-wide_enrichment.pdf), Subhajyoti _et. al._, Briefings in Bioinformatics 2013

[GSuite HyperBrowser: integrative analysis of dataset collections across the genome and epigenome](https://github.com/lexnederbragt/INF-BIOx121/raw/2017/Curriculum/GSuite_HyperBrowser.pdf), Simovski _et. al._, GigaScience 2107

From 2016
============
[Coming of age: ten years of next-generation sequencing technologies](https://wiki.uio.no/projects/clsi/images/6/6f/Goodwin2016.pdf), Goodwin _et. al._, Nature Reviews Genetics 2016

Figures 1-4 from
[Sequencing technologies — the next generation](https://wiki.uio.no/projects/clsi/images/a/a5/Sequencing_technologies_the_next_generation.pdf), Metzker, Nature Reviews Genetics 2009

[An Extensive Evaluation of Read Trimming Effects on Illumina NGS Data Analysis](https://wiki.uio.no/projects/clsi/images/3/3e/Fabbro2013.pdf), Fabbro _et. al._, PLoS ONE 2013

[Sequencing depth and coverage: key considerations in genomic analyses](https://wiki.uio.no/projects/clsi/images/7/75/Sims2014.pdf), Sims _et. al._, Nature Reviews Genetics 2014

[A field guide to whole‐genome sequencing, assembly and annotation](https://wiki.uio.no/projects/clsi/images/9/99/Ekblom2014.pdf), Ekblom and Wolf, Evolutionary Applications 2014

[The Theory and Practice of Genome Sequence Assembly](https://wiki.uio.no/projects/clsi/images/8/8f/Simpson2015.pdf), Simpson and Pop, Annual Review of Genomics and Human Genetics 2015

[Benchmarking short sequence mapping tools](https://wiki.uio.no/projects/clsi/images/e/ea/Hatem2013.pdf) , Hatem  _et. al._, BMC Bioinformatics 2013

[Genotype and SNP calling from next-generation sequencing data](https://wiki.uio.no/projects/clsi/images/3/33/Genotype_and_SNP_calling_from_next-generation_sequencing_data.pdf), Nielsen _et. al._, Nature Reviews Genetics 2011

[A survey of best practices for RNA-seq data analysis](https://wiki.uio.no/projects/clsi/images/a/a6/Conesa2016.pdf), Conesa _et. al._ Genome Biology 2016

[The power and promise of RNA‐seq in ecology and evolution](https://wiki.uio.no/projects/clsi/images/6/63/Todd2016.pdf), Todd _et. al._, Molecular Ecology 2016






